docs/madr/decisions/008-mesh-logging.md

```diff
@@ -0,0 +1,381 @@
+
+For now new Policy `MeshLogging` will reuse the backend definitions from `Mesh`.
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: Mesh
+metadata:
+  name: default
+spec:
+  logging:
+    # MeshLogging policies may leave the `backend` field undefined.
+    # In that case the logs will be forwarded into the `defaultBackend` of that Mesh.
+    defaultBackend: file
+    # List of logging backends that can be referred to by name
+    # from MeshLogging policies of that Mesh.
+    backends:
+      - name: logstash
+        type: tcp
+        conf:
+          address: 127.0.0.1:5000
+      - name: file
+        type: file
+        conf:
+          path: /tmp/access.log
+---
+type: MeshLogging
+mesh: default
+spec:
+  targetRef:
+    kind: MeshService
+    name: web-backend
+  from:
+    - targetRef:
+        kind: MeshService
+        name: web-frontend
+      default:
+        backend: logstash # Forward the logs into the logging backend named `logstash`.
+```
+
+In the future we will introduce `MeshLoggingBackend` policy that would hold that data.
+Definitions there will have precedence over definitions in `Mesh`.
+In the future the user will re-create (or maybe we could do this automatically) equivalent `MeshLoggingBackend` policies.
+
+##### Positive Consequences 
+
+* easier to implement
+
+##### Negative Consequences
+
+* breaks the barrier between new / old policies by reusing old policies resources
+* possibly more work for the user in the future
+
+#### Move backend to the `MeshLoggingBackend` policy now
+
+Moving backend to a new policy called `MeshLoggingBackend` immediately.
+That policy would be neither `inbound` nor `outbound` it would just store backend definitions.
+It would support `targetRef` of `Mesh` but could probably support more in the future.
+This allows us to have consistent naming (`enabledBackend`) across all policies and
+would make it easier to have per zone backends.
+
+```yaml
+type: MeshLoggingBackend
+mesh: default
+spec:
+  targetRef:
+    kind: Mesh
+    name: default
+  enabledBackend: file # instead of defaultBackend
+  # List of logging backends that can be referred to by name
+  # from MeshLogging policies of that Mesh.
+  default:
+    backends:
+      - name: logstash
+        type: tcp
+        conf:
+          address: 127.0.0.1:5000
+      - name: file
+        type: file
+        conf:
+          path: /tmp/access.log
+---
+type: MeshLogging
+mesh: default
+spec:
+  targetRef:
+    kind: MeshService
+    name: web-backend
+  from:
+    - targetRef:
+        kind: MeshService
+        name: web-frontend
+      default:
+        backend: logstash # Forward the logs into the logging backend named `logstash`.
+```
+
+##### Positive Consequences
+
+* cleaner solution, there is separation between old and new policies
+* consistency in regards to `enabledBackend`
+* easier to have per zone backends
+
+##### Negative Consequences
+
+* possibly more work to implement
+
+#### Embed backend in `MeshLogging` policy
```

Comment:
I guess one option is to have it both in the policy and at the mesh level.

You can either do:

```
default:
  backend:
     name: logstash
     type: tcp
     conf: ....
```

Or

```
default:
   backend:
      type: reference
      name: aBackendDefinedInMesh
```

Reply:
I'll add this to possible options.